### PR TITLE
Stop database restore with -o option when no space left on disk

### PR DESCRIPTION
**Pull request by ilya071294**
When this error occurs, gbak tries to restore all remained records. Restore log and firebird.log are growing up to several GB because of errors. The same occurs if connection was lost. I think it's better to allow to continue restore for some errors (currently isc_not_valid and isc_malformed_string), but prohibit for all others.

### CODE REVIEW

**Comment by ilya071294** [comment]
Please, review it.

**Comment by ilya071294** [comment]
There are many cases when -o option might help to restore database with ignoring errors. For this reason gbak should stop restore only when unrecoverable errors occur (such as isc_io_error, isc_att_shutdown, isc_net_write_err).

File: src/burp/restore.epp [hunk 1]
```diff
@@ -3057,7 +3059,10 @@ rec_type get_data(BurpGlobals* tdgbl, burp_rel* relation, bool skip_relation)
 			}
 			else
 			{
-				if (tdgbl->gbl_sw_incremental)
+				if (tdgbl->gbl_sw_incremental &&
```
Comment (AlexPeshkoff) [inline src/burp/restore.epp]: I agree with an idea to make error processing during restore more flexible. But list of fatal errors appears to be incomplete. Except isc_net_write_err isc_net_read_err can happen, except isc_att_shutdown - isc_shutdown, etc. On the other hand all such gds codes are already classified as SQL code -902. Why not use it instead?

**Comment by AlexPeshkoff** [comment]
I see PR to B3_0_Release. Same changes in master are also required.

**Comment by ilya071294** [comment]
If PR is OK now, I'll prepare the same for master.

**Comment by AlexPeshkoff** [comment]
I see no more problems with this PR

**Pull request merged**